src/_guides/language/evolution.md

```diff
@@ -180,131 +183,190 @@ turn into edit-time analysis errors.
 
-Dart 2.13 expanded support for **[type aliases][]** (`typedef`),
-which used to work only for function types
-but now work for any type.
-You can use the new name created with a type alias
-anywhere the original type could be used.
+Dart 2.10 added no new features to the Dart language.
+It did add an expanded [`dart` tool][dart-tool] similar to the
+Flutter SDK's [`flutter` tool][].
 
-Dart 2.13 also improved the struct support in **[Dart FFI][]**,
-adding support for inline arrays and packed structs.
+### Dart 2.9
 
-### Dart 2.14
+Dart 2.9 added no new features to the Dart language.
 
-Dart 2.14 added the unsigned shift operator (`>>>`),
-also known as _triple-shift_.
-This new operator is similar to `>>`,
-except that it always fills the most significant bits with zeros.
-For more information, see the
-[bitwise and shift operator][] section of the language tour.
+### Dart 2.8
+_Released 6 May 2020_
+| [Dart 2.8 announcement](https://medium.com/dartlang/announcing-dart-2-8-7750918db0a)
 
-[bitwise and shift operator]: /language/operators#bitwise-and-shift-operators
+Dart 2.8 didn't add any features to the Dart language. It did
+contain a number of preparatory [breaking changes][2.8 breaking changes]
+to improve nullability-related usability and performance for [null safety][].
 
-Dart 2.14 also removed some restrictions on type arguments.
-You can now pass type arguments to annotations,
-and you can use a generic function type as a type argument.
-All of the following code was invalid before 2.14,
-but is now allowed:
+It improved the performance of the `pub` tool and
+added the [`pub outdated`][] command.
+
+### Dart 2.7
+_Released 11 December 2019_
+| [Dart 2.7 announcement](https://medium.com/dartlang/dart-2-7-a3710ec54e97)
+
+Dart 2.7 added support for **[extension methods][]**,
+enabling you to add functionality to any type
+—-even types you don’t control—-
+with the brevity and auto-complete experience of regular method calls.
+
+The following example extends the `String` class from
+`dart:core` with a new `parseInt()` method:
 
 ```dart
-@TypeHelper<int>(42, "The meaning")
-late List<T Function<T>(T)> idFunctions;
-var callback = [<T>(T value) => value];
-late S Function<S extends T Function<T>(T)>(S) f;
+extension ParseNumbers on String {
+  int parseInt() {
+    return int.parse(this);
+  }
+}
+
+void main() {
+  int i = '42'.parseInt();
+  print(i);
+}
 ```
 
-### Dart 2.15
+### Dart 2.6
+_Released 5 November 2019_
+| [Dart 2.6 announcement](https://medium.com/dartlang/dart2native-a76c815e6baf)
 
-Dart 2.15 improved support for function pointers,
-known as _tear-offs._
-In particular, constructor tear-offs are now supported.
-For details, see the [Dart 2.15 announcement][].
+Dart 2.6 introduced a
+[breaking change (dart-lang/sdk#37985)](https://github.com/dart-lang/sdk/issues/37985).
+Inference changes when using Null values in a `FutureOr` context.
+Constraints of the forms similar to `Null <:FutureOr` now yield `Null`
```

Comment:
The `<:` should be outside of the code.

```suggestion
Constraints of forms similar to `Null` <: `FutureOr` now yield `Null`
```

Also we might want to avoid the symbol completely. It's something readers may not be familiar with. Perhaps something like?

```suggestion
Constraints of the form where `Null` is a subtype of `FutureOr` now yield `Null`
```